core/src/main/java/org/apache/iceberg/actions/BaseCommitService.java

```diff
@@ -169,7 +169,11 @@ public void close() {
 
     if (!completedRewrites.isEmpty() && timeout) {
       LOG.error("Attempting to cleanup uncommitted file groups");
-      completedRewrites.forEach(this::abortFileGroup);
+      synchronized (completedRewrites) {
+        while (!completedRewrites.isEmpty()) {
```

Comment:
the `synchronized` looks good to me. 

regarding the while loop, curious why not stay with the original code?

Reply:
@stevenzwu, we need to poll the `completedRewrites` to empty, otherwise the `committerService ` still could commit them.

Reply:
The issue is we could leave the synchronized block and then touch completedRewrites?

Reply:
@RussellSpitzer that's right.